docs/developer/extending/apps/developing-apps/app-error-codes.mdx

```diff
@@ -0,0 +1,225 @@
+---
+title: App error codes
+---
+
+Below you'll find explanations for `AppErrorCode` values that can be returned during various app-related operations in Saleor, such as installing an app, fetching its manifest, updating an app, or deleting it. Each section explains what the error means, when it can happen, and what you should check or do to resolve it.
+
+## `AppErrorCode.INVALID_URL_FORMAT`
+This error is raised when a URL is malformed or uses an unsupported scheme.
+It can occur for field `manifestUrl` (when fetching app's manifest) or for any URL field within the manifest content
+(such as `tokenTargetUrl`, extension URLs, webhook `targetUrl`, or brand logo URLs).
+
+- 🧐 **Cause**: A URL (either `manifestUrl` or a URL inside the manifest) is invalid, malformed, or uses an unsupported scheme for its context. For example:
+  - `manifestUrl` supports only `http://` or `https://`.
```

Comment:
Are you sure that we always support  `http` / `https`? I think that this is controller by the environment 🤔  In debug mode (local env) `http` works as expected, but drops the `https` manifests. When hosted, we will drop `http` request but accept `https`.
Maybe worth to check this/mention?

Reply:
I've checked it on production with URL: `http://localhost:3000` and we return error code `AppErrorCode.INVALID`, not this one, since Saleor couldn't fetch the host - validation didn't pick this up.

It makes sense because we use:
- `class AppURLValidator(URLValidator)` - which uses Django's URLValidator, which [by default supports `http` protocol](https://docs.djangoproject.com/en/5.2/ref/validators/#django.core.validators.URLValidator.schemes)
- for `targetUrl` we override this to allow awssqs and gcppubsub protocols:

```
    target_url_validator = AppURLValidator(
        schemes=["http", "https", "awssqs", "gcpubsub"]
    )
```

Reply:
@korycins maybe we should change this validation, as you mentioned since it doesn't make sense to connect to local ports on production